logic/tool/TypeResolver.java

```diff
@@ -263,64 +261,75 @@ private TypeVariable createRootTypeVar(Label rootLabel) {
             return newRootType;
         }
 
-        public ResolverVariables getResolveVars() {
-            return resolverVariables;
+        Resolvers resolvers() {
+            return resolvers;
         }
 
-        public HashMap<TypeVariable, Set<TypeVariable>> getVariableNeighbours() {
+        HashMap<TypeVariable, Set<TypeVariable>> neighbours() {
             return neighbours;
         }
 
-        private boolean noInformation(TypeVariable variable) {
-            return !variable.reference().isLabel() && !variable.sub().isPresent() &&
-                    !variable.label().isPresent() && variable.is().isEmpty();
+        private boolean hasNoInfo(TypeVariable variable) {
+            return (!variable.reference().isLabel() &&
+                    !variable.sub().isPresent() &&
+                    !variable.label().isPresent() &&
+                    variable.is().isEmpty());
         }
 
-        private TypeVariable convertVariable(Variable variable) {
-            if (resolverVariables.hasConversion(variable)) return resolverVariables.getConversion(variable);
-            if (variable.isType()) {
-                TypeVariable asTypeVar = resolverVariables.convert(variable);
-                addNeighboursOfTypeVariable(variable.asType(), asTypeVar);
-                neighbours.putIfAbsent(asTypeVar, new HashSet<>());
-            } else convertThingVariable(variable.asThing());
-            return resolverVariables.getConversion(variable);
+        private TypeVariable convert(Variable variable) {
+            if (variable.isType()) return convert(variable.asType());
+            else if (variable.isThing()) return convert(variable.asThing());
+            else throw GraknException.of(ILLEGAL_STATE);
+        }
+
+        private TypeVariable convert(TypeVariable variable) {
+            if (resolvers.contains(variable)) return resolvers.resolver(variable);
+
+            TypeVariable resolver = resolvers.register(variable);
+            copyNeighbours(variable, resolver);
+            // TODO: Why do we need `neighbours.putIfAbsent(resolver, new HashSet<>());` ?
+            //       Is this a patch for the fact that we forgot to use `to` variable in `copyNeighbours(from, to)` ?
+            neighbours.putIfAbsent(resolver, new HashSet<>());
```

Comment:
We can now see more clearly what the we do with `TypeVariable` by putting the logic into its own method. However, I'm still not sure why we need `neighbours.putIfAbsent(resolver, new HashSet<>());`. Is this a patch for the fact that we forgot to use `to` variable in `copyNeighbours(from, to)` ?

Reply:
`neighbours.putIfAbsent(resolver, new HashSet<>());` was necessary. But it's a moot point since the new `TypeResovler` will always be exhaustive, and therefore not need to know neighbourhood distance.